zmon_agent/discovery/kubernetes/cluster.py

```diff
@@ -453,28 +455,55 @@ def get_cluster_postgresdbs(kube_client, cluster_id, alias, region, infrastructu
 
         service_namespace = obj['metadata']['namespace']
         service_dns_name = '{}.{}.svc.cluster.local'.format(service.name, service_namespace)
-        service_port = obj['spec']['ports'][0]['port']  # Assume the first port is the one we need.
 
-        dbnames = list_postgres_databases(host=service_dns_name,
-                                          port=service_port,
+        entity = {
+            'id': '{}-{}[{}]'.format(service.name, service.namespace, cluster_id),
+            'type': POSTGRESQL_CLUSTER_TYPE,
+            'kube_cluster': cluster_id,
+            'alias': alias,
+            'created_by': AGENT_TYPE,
+            'infrastructure_account': infrastructure_account,
+            'region': region,
+
+            'dnsname': service_dns_name,
+            'shards': {
+                'postgres': '{}:{}/postgres'.format(service_dns_name, POSTGRESQL_DEFAULT_PORT)
+            }
+        }
+
+        entities.append(entity)
+
+    return entities
+
+
+def get_postgresql_databases(postgresql_clusters, cluster_id, alias, region, infrastructure_account,
+                             postgres_user, postgres_pass):
+    if not (postgres_user and postgres_pass):
+        return []
+
+    entities = []
+
+    for pgcluster in postgresql_clusters:
+        dbnames = list_postgres_databases(host=pgcluster['dnsname'],
+                                          port=POSTGRESQL_DEFAULT_PORT,
                                           user=postgres_user,
                                           password=postgres_pass,
                                           dbname='postgres',
                                           sslmode='require')
         for db in dbnames:
             entity = {
-                'id': '{}-{}-{}[{}]'.format(db, service.name, service.namespace, cluster_id),
-                'type': POSTGRESDB_TYPE,
+                'id': '{}-{}'.format(db, pgcluster['id']),
```

Comment:
Correct.  We generate the postgresql cluster ids with proper kube cluster id suffix.  Pleas see in `get_postgresql_clusters()` above.